extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    internal static class EventGridHelper
+    {
+        internal static ValueTask<ConversionResult> DeserializeToTargetType(ConverterContext context)
+        {
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Failed(new InvalidOperationException("Context source cannot be null")));
+                }
+                
+                var targetType = context!.TargetType;
+                var item = JsonSerializer.Deserialize(contextSource, targetType);
+                return new(ConversionResult.Success(item));
+            }
+            catch (JsonException ex)
+            {
+                string msg = String.Format(CultureInfo.CurrentCulture,
+                    @"Binding parameters to complex objects uses JSON serialization.
+                    1. Bind the parameter type as 'string' instead to get the raw values and avoid JSON deserialization, or
+                    2. Change the event payload to be valid json.
+                    The JSON parser failed: {0}",
```

Comment:
The exception is passed in as the inner exception. Should be fine to remove the message.